frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +537,138 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() != '.' &&
+         "expected no period at end of ErrorParsing message");
+  errorMessage.push_back('.');
```

Comment:
Instead of this, could we just assert that there _is_ a period and modify the call sites to have the periods? That would be consistent with the approach in other PRs. As we go, I think the goal is to eventually to be strict about what error messages people write / report, and rely less on special logic in `ErrorBase` and friends to fix up punctuation.